cmd/minikube/cmd/start.go

```diff
@@ -144,6 +144,9 @@ func runStart(cmd *cobra.Command, args []string) {
 		registryMirror = viper.GetStringSlice("registry_mirror")
 	}
 
+	if !config.ProfileNameValid(ClusterFlagValue()) {
+		exit.WithCodeT(exit.Config, "Profile name {{.name}} is not valid, only '[a-zA-Z0-9][a-zA-Z0-9_.-]' are allowed", out.V{"name": ClusterFlagValue()})
```

Comment:
Please use a more user friendly message, so users don't have to decrypt the regex.

Reply:
What about something like this?

Profile name {{.name}} is not valid. It should be minimum 2 characters long, only alphanumeric characters, dots, underscores and dashes '-' are permitted, starting by alphanumeric.

It's a bit longer though.